porting/npl/linux/src/os_atomic.c

```diff
@@ -22,10 +22,21 @@
 
 #include "nimble/nimble_npl.h"
 
+#ifdef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
 static pthread_mutex_t s_mutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+#else
+static pthread_mutex_t s_mutex = PTHREAD_MUTEX_INITIALIZER;
+static uint8_t s_mutex_inited = 0;
```

Comment:
I am runing it on an openwrt system,  found a nested enter_critical() function call as below. When i use non-recursive mutex, it hangs there.
btw, since the wrapper function named OS_ENTER_CRITICAL(),  nested call should be allowed ?

```c
ble_hci_sock_rx_msg(void)
{
	505: OS_ENTER_CRITICAL(sr);
            rc = ble_hci_sock_rx_cmd_cb(data, ble_hci_sock_rx_cmd_arg); //==> call ble_hs_event_rx_hci_ev()
	    OS_EXIT_CRITICAL(sr); 
}

static void ble_hs_event_rx_hci_ev(struct ble_npl_event *ev)
{   
    //...
    rc = os_memblock_put(&ble_hs_hci_ev_pool, ev); //==> call os_memblock_put_from_cb()
    BLE_HS_DBG_ASSERT_EVAL(rc == 0);
    //...
}

os_error_t os_memblock_put_from_cb(struct os_mempool *mp, void *block_addr)
{
    //...
    block = (struct os_memblock *)block_addr;
    OS_ENTER_CRITICAL(sr);
    //...
    OS_EXIT_CRITICAL(sr);    
    return OS_OK;
}
```